src/classrooms/classrooms.service.ts

```diff
@@ -65,4 +65,17 @@ export class ClassroomsService {
     // TODO: query for fetching classrooms that a student is enrolled in.
     return classrooms;
   }
+
+  async getClassroomById(id: number, userId: number): Promise<Classroom | null> {
+    const user = await this.em.findOneOrFail(User, { id: userId }, { populate: ["teacher"] });
+
+    const classroom = await this.em.findOneOrFail(Classroom, { id }, { populate: ["teacher"] });
+
+    if (user.teacher && classroom.teacher.id === user.teacher.id) {
+      return classroom;
+    }
+    // TODO: Add check for students enrolled in the classroom
+    // For now, throw ForbiddenException if the user is not the teacher
+    throw new ForbiddenException("You do not have permission to access this classroom");
```

Comment:
The service should always return a classroom and exceptions should be thrown based on some condition.
I think this implementation is flipped.